src/frontend/xgboost/xgboost_json.h

```diff
@@ -0,0 +1,376 @@
+  bool StartArray();
+
+  bool StartObject();
+
+  bool Uint(unsigned u);
+
+  bool EndObject(std::size_t memberCount);
+
+ private:
+  std::vector<double> loss_changes;
+  std::vector<double> sum_hessian;
+  std::vector<double> base_weights;
+  std::vector<int> leaf_child_counts;
+  std::vector<int> left_children;
+  std::vector<int> right_children;
+  std::vector<int> parents;
+  std::vector<int> split_indices;
+  std::vector<double> split_conditions;
+  std::vector<bool> default_left;
+  int num_nodes;
+};
+
+/*! \brief handler for GBTreeModel objects from XGBoost schema*/
+class GBTreeModelHandler : public OutputHandler<treelite::Model> {
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool StartArray();
+  bool StartObject();
+};
+
+/*! \brief handler for GradientBoosterHandler objects from XGBoost schema*/
+class GradientBoosterHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool Uint(unsigned u);
+  bool String(const char *str, std::size_t length, bool copy);
+  bool StartObject();
+};
+
+/*! \brief handler for ObjectiveHandler objects from XGBoost schema*/
+class ObjectiveHandler : public OutputHandler<std::string> {
+  using OutputHandler<std::string>::OutputHandler;
+
+  bool StartObject();
+
+  bool String(const char *str, std::size_t length, bool copy);
+};
+
+/*! \brief handler for LearnerParam objects from XGBoost schema*/
+class LearnerParamHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool String(const char *str, std::size_t length, bool copy);
+};
+
+/*! \brief handler for Learner objects from XGBoost schema*/
+class LearnerHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool StartArray();
+  bool StartObject();
+  bool EndObject(std::size_t memberCount);
+
+ private:
+  std::string objective;
+};
+
+/*! \brief handler for XGBoostModel objects from XGBoost schema*/
+class XGBoostModelHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool StartArray();
+  bool StartObject();
+  bool EndObject(std::size_t memberCount);
+
+ private:
+  std::vector<unsigned> version;
+};
+
+/*! \brief handler for root object of XGBoost schema*/
+class RootHandler : public OutputHandler<treelite::Model> {
+ public:
+  using OutputHandler<treelite::Model>::OutputHandler;
+  bool StartObject();
+};
+
+/*! \brief handler which delegates JSON parsing to stack of delegates*/
+class DelegatedHandler
+    : public rapidjson::BaseReaderHandler<rapidjson::UTF8<>, DelegatedHandler>,
+      public Delegator {
+
+ public:
+  /*! \brief create DelegatedHandler with initial RootHandler on stack */
+  static std::shared_ptr<DelegatedHandler> create() {
+    struct make_shared_enabler : public DelegatedHandler {};
+
+    std::shared_ptr<DelegatedHandler> new_handler = \
```

Comment:
```suggestion
    std::shared_ptr<DelegatedHandler> new_handler =
```
Backslash not needed here.